Exercise_4.java

```diff
@@ -32,6 +34,27 @@ static void inorder(Node temp)
     /*function to insert element in binary tree */
     static void insert(Node temp, int key) 
     { 
+    	Queue<Node> q = new LinkedList<Node>();
+    	
+    	while(!q.isEmpty()) {
+    		Node current = q.poll();
+    		
+    		if(current.left == null) {
+    			current.left = new Node(key);
+    		}
+    		else {
+    			q.add(current.left);
+    		}
+    		if(current.right == null) {
+    			current.right = new Node(key);
+    		}
+    		else {
+    			q.add(current.right);
+    			

```

Comment:
After adding in both else statements, the loop needs to break. Otherwise the node will be added multiple times at where ever it encounters a null.